File: backend/src/WebApi/Models/ExpenseModels.cs
namespace WebApi.Models;

public sealed record ExpenseInfo(
    Guid Id,
    string Description,
    decimal Amount,
    DateTime Date,
    string? Category,
    Guid UserId,
    string UserName,
    string? BillingCompany,
    string? BillingStreet,
    string? BillingPostalCode,
    string? BillingCity);

public sealed record CreateExpenseRequest(
    string Description,
    decimal Amount,
    DateTime Date,
    string? Category,
    Guid UserId,
    string? BillingCompany,
    string? BillingStreet,
    string? BillingPostalCode,
    string? BillingCity);

public sealed record ExpenseReport(
    Guid UserId,
    string UserName,
    int Year,
    int Month,
    DateTime PeriodStart,
    DateTime PeriodEnd,
    decimal TotalAmount,
    int ExpenseCount,
    IEnumerable<ExpenseInfo> Expenses);

File: frontend/libs/data-access/src/lib/models/expense.models.ts
export interface Expense {
  id: string;
  description: string;
  amount: number;
  date: string;
  category?: string;
  userId: string;
  userName: string;
  billingCompany?: string;
  billingStreet?: string;
  billingPostalCode?: string;
  billingCity?: string;
}

export interface CreateExpenseRequest {
  description: string;
  amount: number;
  date: string;
  category?: string;
  userId: string;
  billingCompany?: string;
  billingStreet?: string;
  billingPostalCode?: string;
  billingCity?: string;
}

export interface ExpenseReport {
  userId: string;
  userName: string;
  year: number;
  month: number;
  periodStart: string;
  periodEnd: string;
  totalAmount: number;
  expenseCount: number;
  expenses: Expense[];
}

